src/main.rs

```diff
@@ -489,9 +491,14 @@ impl Halloy {
                     let commands = messages
                         .into_iter()
                         .flat_map(|message| {
+                            let events = match self.clients.receive(&server, message) {
+                                Ok(events) => events,
+                                Err(e) => return vec![Task::done(Message::IrcError(e))],
```

Comment:
We shouldn't be returning a `Task` / `Message` here for simply logging the error, we can handle this in the current code path. Let's just instead have a function called `log_irc_error` or something and call that directly

Reply:
I had a task initially for the modal, but IMO we should keep it just in case the behaviour changes in the future (like with a red dot getting shown)

Reply:
Let's not keep dead code around in anticipation of something that may never land. This current pattern is not one we want to use.

Reply:
sounds good :)